hypervisor/src/arch/x86/mod.rs

```diff
@@ -34,3 +20,44 @@ pub const MTRR_MEM_TYPE_WB: u64 = 0x6;
 
 // IOAPIC pins
 pub const NUM_IOAPIC_PINS: usize = 24;
+
+macro_rules! msr {
+    ($msr:expr) => {
+        MsrEntry {
+            index: $msr,
+            data: 0x0,
+            ..Default::default()
+        }
+    };
+}
+#[cfg(feature = "kvm")]
+macro_rules! msr_data {
+    ($msr:expr, $data:expr) => {
+        MsrEntry {
+            index: $msr,
+            data: $data,
+            ..Default::default()
+        }
+    };
+}
+
+pub fn boot_msr_entries() -> MsrEntries {
```

Comment:
Not sure that's the right approach here. If we want this to be available to hyper-v, we need to define the same thing in `hypervisor/src/hyperv/x86_64/mod.rs`. I know they might be a bit of duplicate, but we can't assume `MsrEntries` and `MsrEntry` structures are going to be identical between KVM and Hyper-V.

Reply:
My thought as well.

Reply:
Oh that's cool, but still I think we shouldn't take shortcuts since it does not make a proper abstraction layer otherwise.

Reply:
Okay. I will drop the PR then. Right now we have separate implementation for each hypervisor in our internal branch.